redaxo/src/addons/users/lib/role.php

```diff
@@ -35,14 +35,28 @@ class rex_user_role implements rex_user_role_interface
      *
      * @param array $params Params
      */
-    private function __construct(array $params)
+    private function __construct(array $roles)
     {
-        foreach ([rex_perm::GENERAL, rex_perm::OPTIONS, rex_perm::EXTRAS] as $key) {
-            $perms = $params[$key] ? explode('|', trim($params[$key], '|')) : [];
-            $this->perms = array_merge($this->perms, $perms);
-            unset($params[$key]);
+        foreach ($roles as $role) {
+            $params = $role;
```

Comment:
wozu diese Variablenumbenennung? Warum nicht mit `$role` weiterarbeiten? oder direkt in `foreach` die Variable `$params` nennen..